mlir/lib/Dialect/Mesh/IR/MeshOps.cpp

```diff
@@ -471,46 +460,42 @@ LogicalResult verifyReduceScatterOperandAndResultShape(
 //===----------------------------------------------------------------------===//
 
 LogicalResult
-mlir::mesh::AllReduceOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+AllReduceOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
   return verifyMeshSymbolUses(*this, symbolTable);
 }
 
-void mlir::mesh::AllReduceOp::getCanonicalizationPatterns(
-    RewritePatternSet &patterns, MLIRContext *context) {
-  populateMeshAxesSetCanonicalizationPatterns<AllReduceOp>(patterns, context);
+LogicalResult mlir::mesh::AllReduceOp::verify() {
+  return verifyMeshAxes(getLoc(), getMeshAxes());
```

Comment:
We should verify the shape of all_reduce.

Reply:
It has the trait `SameOperandsAndResultShape`.

Reply:
Never mind, I actually do the check in `verifyMeshSymbolUses`.

Reply:
Do you mean this check?
```
if (!meshAxes) {
    return success();
  }
```

But an ArrayAttr with an empty array is not nullptr.

Reply:
I removed this check since I made the `mesh_axes` attribute non-optional. It still has a default value.